uk.co.compucorp.civicrm.tasksassignments/js/test/crm-tasks-workflows/controllers/casetype.extended.controller.spec.js

```diff
@@ -1,44 +1,92 @@
 /* eslint-env jasmine */
+/* global angular */
 
 (function () {
   'use strict';
 
   describe('CaseTypeExtendedController', function () {
-    var $controller, $rootScope, $scope, crmApi, activityOptionsData, activityOptions;
+    var $controller, $log, $rootScope, $scope, TaskActivityOptionsData, DocumentActivityOptionsData;
 
     beforeEach(function () {
       module('crm-tasks-workflows.mocks', 'crm-tasks-workflows.controllers');
     });
 
-    beforeEach(inject(function (_$controller_, _$rootScope_, _activityOptionsData_) {
+    beforeEach(inject(function (_$controller_, _$log_, _$rootScope_, _TaskActivityOptionsData_, _DocumentActivityOptionsData_) {
       $controller = _$controller_;
+      $log = _$log_;
       $rootScope = _$rootScope_;
-      activityOptionsData = _activityOptionsData_;
+      DocumentActivityOptionsData = _DocumentActivityOptionsData_;
+      TaskActivityOptionsData = _TaskActivityOptionsData_;
 
-      activityOptions = { values: activityOptionsData };
-
-      initController(crmApi, activityOptions);
+      spyOn($log, 'debug');
+      initController();
     }));
 
-    it('returns the mapped data from api', function () {
-      var expectedValues = activityOptionsData.map(function (type) {
-        return { id: type.name, text: type.label, icon: type.icon };
+    describe('when initialized', function () {
+      var expectedValue;
+
+      beforeEach(function () {
+        expectedValue = { actTypes: { values: getActivityTypes() } };
       });
 
-      expect($scope.activityTypeOptions).toEqual(expectedValues);
+      it('initialises the parent controller with activity types having component labels', function () {
+        expect($log.debug).toHaveBeenCalledWith('Init CaseTypeCtrl with', jasmine.any(Object), jasmine.any(Object), expectedValue);
+      });
     });
 
-    function initController (crmApi, activityOptions) {
-      $scope = $rootScope.$new();
+    describe('addActivity', function () {
+      var activityTypes, activityType;
+      var addActivityInParent = jasmine.createSpy('addActivityInParent');
+
+      beforeEach(function () {
+        $scope = $rootScope.$new();
+        $scope.addActivity = addActivityInParent;
+
+        activityTypes = getActivityTypes();
+        activityTypes[0].reference_activity = '1';
+        activityType = activityTypes[0].name;
+
+        initController($scope);
+        $scope.addActivity({ activityTypes: activityTypes }, activityType);
+      });
+
+      it('calls the parent function', function () {
+        expect(addActivityInParent).toHaveBeenCalledWith({ activityTypes: activityTypes }, activityType);
+      });
+
+      it('removes the reference property for the activity type', function () {
+        expect(activityTypes[0].reference_activity).toBeUndefined();
```

Comment:
Did you confirm if removing this property doesn't mess up creating a new case?

The `reference_activity` along with the `offset` are used to determine the date for the activity. Ex: activity "Schedule joining date", reference activity is "Open Case" (first activity), offset is 3, date of the case is "2020-01-01", then the date for the the "Schedule joining date" will be "2020-01-04".

Can you confirm that still works?

* Go to  `/civicrm/tasksassignments/dashboard#/tasks` 
* Create a new workflow
* Check if the dates for each task are set properly.

Thanks.

Reply:
yeah it works.

![2018-04-18 at 7 01 pm](https://user-images.githubusercontent.com/5058867/38935073-fecc7e44-433a-11e8-9e60-1e14a246f231.png)

![2018-04-18 at 7 02 pm](https://user-images.githubusercontent.com/5058867/38935091-09a719d2-433b-11e8-9cc2-2d31925d97a0.png)

Reply:
Thanks for checking